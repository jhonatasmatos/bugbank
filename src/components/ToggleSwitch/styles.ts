import styled, { css } from 'styled-components';

type Props = {
  isChecked: boolean;
};

export const Container = styled.label`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.primary};
    width: 5rem;
    height: 2rem;
    border-radius: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    position: relative;
    background: ${theme.colors.gray};

    ${(p: Props) =>
      p.isChecked &&
      css`
        background: ${theme.colors.primary};
      `}
  `}
`;

export const Input = styled.input`
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
`;

export const Span = styled.span`
  ${({ theme }) => css`
    position: absolute;
    width: 2rem;
    height: 2rem;
    border-radius: 1rem;
    background: ${theme.colors.white};
    left: 0;
    cursor: pointer;
    box-shadow: 2px 2px 4px ${theme.colors.primary};
    transition: all 0.3s;

    ${(p: Props) =>
      p.isChecked &&
      css`
        left: 60%;
      `}
  `}
`;
