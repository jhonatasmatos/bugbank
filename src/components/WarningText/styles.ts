import styled, { css } from 'styled-components';

interface IWarnigWrapper {
  color?: 'white' | 'secondary';
}

export const WarnigWrapper = styled.p`
  ${({ theme }) => css`
    padding: 1.6rem;
    border: 1px solid ${theme.colors.secondary};
    border-radius: 0.8rem;
    color: ${theme.colors.secondary};
    font-style: italic;
    font-size: 1.3rem;
    text-align: center;
    line-height: 1.6rem;

    span {
      font-weight: bold;
    }

    ${(p: IWarnigWrapper) =>
      p.color === 'white' &&
      css`
        border: 1px solid ${theme.colors.white};
        color: ${theme.colors.white};
      `}
  `}
`;
