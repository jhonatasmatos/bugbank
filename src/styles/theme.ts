export const theme = {
  colors: {
    primary: '#D62A92',
    secondary: '#A422E3',
    tertiary: '#E224ED',
    gray: '#d3d3d3',
    black: '#171717',
    white: '#fff',
  },
  font: {
    family: {
      default: "'Nunito', sans-serif",
    },
  },
  sizes: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
