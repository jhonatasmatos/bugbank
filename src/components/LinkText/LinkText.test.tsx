import { fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { LinkText } from '.';

const fn = jest.fn();

describe('<Modal />', () => {
  it('should render LinkText with text "text link" and href', () => {
    const { container, getByText, debug } = renderTheme(
      <LinkText href="/test">text link</LinkText>,
    );
    debug();

    const linkText = getByText('text link');

    expect(container).toHaveTextContent('text link');
    expect(linkText).toHaveAttribute('href', '/test');
  });

  it('should test click on link', () => {
    const { getByText } = renderTheme(
      <LinkText href="/test" onClick={fn}>
        text link
      </LinkText>,
    );

    const linkText = getByText('text link');

    fireEvent.click(linkText);

    expect(linkText).toHaveAttribute('href', '/test');
    expect(fn).toBeCalledTimes(1);
  });

  it('should render with checked false', () => {
    const { container } = renderTheme(<LinkText>text link</LinkText>);
    expect(container).toMatchSnapshot();
  });
});
