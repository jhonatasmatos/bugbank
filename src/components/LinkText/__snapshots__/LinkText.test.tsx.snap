// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render with checked false 1`] = `
.c0 {
  font-size: 2rem;
  color: #A422E3;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.8;
}

@media (max-width:460px) {
  .c0 {
    font-size: 1.6rem;
  }
}

<div>
  <a
    class="c0"
  >
    text link
  </a>
</div>
`;
